File: api/base/src/main/java/net/corda/v5/base/annotations/Suspendable.java
package net.corda.v5.base.annotations;

import java.lang.annotation.Documented;
import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import static java.lang.annotation.ElementType.METHOD;
import static java.lang.annotation.ElementType.TYPE;
import static java.lang.annotation.RetentionPolicy.RUNTIME;

/**
 * Marks a method as suspendable.
 * <p>
 * This annotation is required to allow a fiber (a special lightweight thread used by flows in Corda) to suspend and
 * release the underlying thread to another fiber.
 */
@Target({ TYPE, METHOD })
@Retention(RUNTIME)
@Documented
public @interface Suspendable {
}

File: base/src/main/kotlin/net/corda/v5/base/annotations/Suspendable.kt
package net.corda.v5.base.annotations

/**
 * Marks a method as suspendable.
 *
 * This annotation is required to allow a fiber (a special lightweight thread used by flows in Corda) to suspend and release the underlying
 * thread to another fiber.
 */
@Target(
    AnnotationTarget.CLASS,
    AnnotationTarget.FUNCTION,
    AnnotationTarget.PROPERTY_GETTER,
    AnnotationTarget.PROPERTY_SETTER,
    AnnotationTarget.PROPERTY,
)
@Retention
@MustBeDocumented
annotation class Suspendable
